### PR TITLE
Add support for VCluster pause/resume

### PR DESCRIPTION
**Pull request by TylerGillson**
Requires:
- https://github.com/spectrocloud/vcluster/pull/3

Related:
- https://github.com/spectrocloud/palette/pull/1807

### CODE REVIEW

File: vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+	}
+
+	for _, item := range list.Items {
+		if item.Annotations == nil || item.Annotations[constants.PausedAnnotation] != "true" {
+			return false, nil
+		}
+
+		originalObject := item.DeepCopy()
+
+		replicas := 1
```
Comment (tiwarisumit3) [inline vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go:294]: If the replica was a different number before pause, shouldn't we bring up the same count?

File: vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+		if item.Annotations[constants.PausedReplicasAnnotation] != "" {
```
Comment (tiwarisumit3) [inline vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go:250]: If the replica was a different number before pause, shouldn't we bring up the same count?

File: vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go (hunk 1 above)
Comment (TylerGillson) [reply vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go:294]: Yes! If you look a few lines down you'll see that the old replica count is retrieved from an annotation. 1 is just the default.

File: vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go (hunk 2 above)
Comment (TylerGillson) [reply vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go:250]: Yes! If you look a few lines down you'll see that the old replica count is retrieved from an annotation. 1 is just the default.

File: vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go (hunk 1 above)
Comment (tiwarisumit3) [reply vendor/github.com/loft-sh/vcluster/pkg/lifecycle/lifecycle.go:294]: sorry, missed to see it previously.

**Review by tiwarisumit3 — APPROVED** [review]

**Pull request merged**